packages/plexus/src/DirectedGraph/builtins/PureNodes.tsx

```diff
@@ -18,15 +16,16 @@ import * as React from 'react';
 
 import Node from './Node';
 
-import type { LayoutVertex, Vertex } from '../../types/layout';
+import { TPropsFactoryFn } from '../types';
+import { TLayoutVertex, TVertex } from '../../types/layout';
 
 type Props = {
-  classNamePrefix: string,
-  getNodeLabel: ?(Vertex) => string | React.Node,
-  layoutVertices: ?(LayoutVertex[]),
-  setOnNode: ?(Vertex) => ?Object,
-  vertexRefs: { current: HTMLElement | null }[],
-  vertices: Vertex[],
+  classNamePrefix: string;
+  getNodeLabel: ((vertex: TVertex) => React.ReactNode) | void;
+  layoutVertices: TLayoutVertex[] | void;
```

Comment:
more of a question than a change suggestion:
when would we want to have this instead of `layoutVertices?: TLayoutVertex[]`?

Reply:
If the prop is provided in all cases, then it should be `layoutVertices:`. If the prop is sometimes omitted, then it should be `layoutVertices?:`